org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/scopes/BuildScopeUnit.scala

```diff
@@ -0,0 +1,90 @@
+package org.scalaide.core.internal.project.scopes
+
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.runtime.SubMonitor
+import org.scalaide.core.IScalaPlugin
+import org.scalaide.core.IScalaProject
+import org.scalaide.core.SdtConstants
+import org.scalaide.core.internal.builder.BuildProblemMarker
+import org.scalaide.core.internal.project.UnitBuilder
+import org.scalaide.ui.internal.preferences.ScalaPluginSettings
+import org.scalaide.util.internal.SettingConverterUtil
+import org.scalaide.core.internal.project.CompileScope
+import org.eclipse.core.resources.IMarker
+import org.eclipse.jdt.core.compiler.IProblem
+import org.eclipse.jdt.ui.text.java.IProblemLocation
+import org.eclipse.jdt.core.IJavaProject
+import org.eclipse.jdt.core.IJavaModelMarker
+
+case object CompileMacrosScope extends CompileScope {
+  def name = "Macros"
+}
+case object CompileMainScope extends CompileScope {
+  def name = "Main"
+}
+case object CompileTestsScope extends CompileScope {
+  def name = "Tests"
+}
+
+abstract class BuildScopeUnit(val owningProject: IScalaProject, val unitBuilder: UnitBuilder,
```

Comment:
As I mentioned before, I'd like to keep far away from `unit` in this part of the code. We already have Compilation Units, and they are essentially source files (both for Scala and Java). `BuildScope` sounds like a good name.